pyccel/codegen/printing/fcode.py

```diff
@@ -2697,12 +2828,41 @@ def split_pos_code(line, endpos):
         result = []
         trailing = ' &'
         for line in lines:
-            if len(line)>72 and ('"' in line[72:] or "'" in line[72:] or '!' in line[:72]):
+            if len(line) > 72 and '!' in line[:72]:
                 result.append(line)
-
-            elif len(line)>72:
+            elif len(line) > 72 and ('"' in line[72:] or "'" in line[72:]):
```

Comment:
Is this used given that you are handling the same case with the `split_long_argument` function? Do you need both?

Reply:
Yes, the `split_long_argument` only handles the arguments of the write call. The lines that are handled by this include the whole expression, But it is not enough on its own because we cannot guarantee that we will always have a space character to perform the split, we could have something like this:
`write(*, (A A A), advance="no") 'hi', 'looong string', 'bye`
the only suitable cuts here will be inside the long string literal expression, thus inside quotes so such a cut will cause problems. That's when the  `split_long_argument`  comes into play and provide the sought space character after splitting the 'looong string'

Reply:
I am not sure i follow, `split_long_argument` only acts on string literals involved in print statements. variable names should not be too long to encompass all the possible splitting points in a line that needs splitting. Not only that but even if that were the case and we had to split the variable name  ie `.... , variable, ....` got split to `...., vari& \n able, ....` this wont be a fatal problem since the & is not inside quotes.  `split_long_argument` sole role is to break string literals since `wrap_fortran` can't.

Reply:
I am not talking about names but actual variables, for example:
```
a = 'very long string'
```

Reply:
> `split_long_argument` sole role is to break string literals since `wrap_fortran` can't.

This is what is causing my question. The line above is the start of the block which handles string literals and is in `wrap_fortran`. So when is this used?